### ISSUE TITLE
[Feature request] cjs config extension support

### ISSUE THREAD

**Issue by taep96** [francoismassart/eslint-plugin-tailwindcss#190]
In projects that have `"type": "module"` set in `package.json`, you have to use `tailwind.config.cjs` instead of `tailwind.config.js`, since TainwindCSS config doesn't support ES modules. Can you also check for `.cjs` config files by default, so there is no need to set the option?

**Comment by ChristianIvicevic** [comment]
This would be marvelous. It took me some time to understand why the plugin would report false positives after migrating to `.cjs`. Right now I have to repeat myself in my custom config as follows:
```js
// This mixin applies some additional checks for projects using Tailwind.
module.exports = {
  plugins: [
    // Plugin documentation: https://www.npmjs.com/package/eslint-plugin-tailwindcss
    'eslint-plugin-tailwindcss'
  ],
  overrides: [
    {
      // Declare an override that applies to TypeScript files only
      files: ['*.ts', '*.tsx'],
      rules: {
        'tailwindcss/classnames-order': ['warn', { config: 'tailwind.config.cjs' }],
        'tailwindcss/enforces-negative-arbitrary-values': 'warn',
        'tailwindcss/enforces-shorthand': 'warn',
        'tailwindcss/migration-from-tailwind-2': 'warn',
        'tailwindcss/no-arbitrary-value': 'off',
        'tailwindcss/no-custom-classname': ['warn', { config: 'tailwind.config.cjs' }],
        'tailwindcss/no-contradicting-classname': 'error'
      }
    }
  ]
};
```

**Comment by taep96** [comment]
You can actually specify your config file in the top level `settings` option. Also consider extending from the recommended config which has the exact same rules set as your's.

Example .eslintrc:
```json
{
    "extends": ["plugin:tailwindcss/recommended"],
    "settings": {
        "tailwindcss": {
            "config": "./tailwind.config.cjs"
        }
    },
    "rules": {
        // Other options here
    }
}
```

**Comment by francoismassart** [comment]
@taep96 & @ChristianIvicevic 
Did you try this issue against the 3.12.0 or 3.12.1 ?
It might be fixed. 
please let me know.
